docker/docker_image_dest.go

```diff
@@ -479,10 +479,10 @@ func (d *dockerImageDestination) PutSignatures(ctx context.Context, signatures [
 		return err
 	}
 	switch {
-	case d.c.signatureBase != nil:
-		return d.putSignaturesToLookaside(signatures, *instanceDigest)
 	case d.c.supportsSignatures:
 		return d.putSignaturesToAPIExtension(ctx, signatures, *instanceDigest)
+	case d.c.signatureBase != nil:
+		return d.putSignaturesToLookaside(signatures, *instanceDigest)
 	default:
```

Comment:
This is now unreachable; consistently with `SupportsSignatures`, either assume `d.c.signatureBase` is non-`nil`, or report an internal error.